library/src/scala/util/ChainingOps.scala

```diff
@@ -37,10 +41,7 @@ final class ChainingOps[A](private val self: A) extends AnyVal {
     *  @tparam U     the result type of the function `f`.
     *  @return       the original value `self`.
     */
-  def tap[U](f: A => U): A = {
-    f(self)
-    self
-  }
+  def tap[U](f: A => U): A = scala.util.chaining.tap(self)(x => f(x))
```

Comment:
```suggestion
  def tap[U](f: A => U): A = scala.util.chaining.tap(self)(f)
```

But it is blocked by #24726